packages/app/src/cli/services/context.ts

```diff
@@ -562,14 +553,60 @@ async function fetchDevDataFromOptions(
   return {app: selectedApp, store: selectedStore}
 }
 
+interface AppDevCachedContext {
+  configuration: AppConfiguration
+  configurationPath: string
+  cachedInfo?: CachedAppInfo
+  remoteApp?: OrganizationApp
+}
+
 /**
- * Retrieve cached info from the global configuration based on the current local app
- * @param reset - Whether to reset the cache or not
+ * Retrieve app info from the cache or the current configuration.
+ * @param reset - Whether to reset the cache or not.
  * @param directory - The directory containing the app.
+ * @param token - The partners token.
  */
-function getAppDevCachedInfo({reset, directory}: {reset: boolean; directory: string}): CachedAppInfo | undefined {
+async function getAppDevCachedContext({
+  reset,
+  directory,
+  token,
+  config,
+}: {
+  reset: boolean
+  directory: string
+  token: string
+  config?: string
+}): Promise<AppDevCachedContext> {
   if (reset) clearAppInfo(directory)
-  return getAppInfo(directory)
+
+  let cachedInfo = getAppInfo(directory)
+
+  const {configuration, configurationPath} = await loadAppConfiguration({
+    directory,
+    configName: config || cachedInfo?.configFile,
+  })
+
+  let remoteApp
+  if (isCurrentAppSchema(configuration)) {
+    remoteApp = await appFromId(configuration.client_id, token)
```

Comment:
IIRC we do that because we're not storing the organization id in the app toml, and we need that later. I've asked if we could include that in the toml as well, but got some pushback :)